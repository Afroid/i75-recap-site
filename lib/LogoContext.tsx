"use client";
import { createContext, useContext, useMemo } from "react";

const logos = [
  "/default-logo.png",
  "/default-logo-alt.png",
  "/default-RWB.png",
];

const LogoContext = createContext<string | null>(null);

export function LogoProvider({ children }: { children: React.ReactNode }) {
  // Randomizes only once on initial render
  const logo = useMemo(
    () => logos[Math.floor(Math.random() * logos.length)],
    []
  );
  return (
    <LogoContext.Provider value={logo}>
      {children}
    </LogoContext.Provider>
  );
}

export function useLogo() {
  const logo = useContext(LogoContext);
  if (logo === null) throw new Error("useLogo must be inside LogoProvider");
  return logo;
}
