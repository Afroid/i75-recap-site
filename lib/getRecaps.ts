import recap2024 from "@/data/recaps/2024.json";
import recap2023 from "@/data/recaps/2023.json";
import { RecapData } from "@/types/types";

export function getRecaps(): Record<number, number[]> {
  const allData: RecapData[] = [recap2024, recap2023]; // Add more as needed
  const summary: Record<number, number[]> = {};

  for (const season of allData) {
    summary[season.year] = season.recaps.map((r) => r.week);
  }

  return summary;
}
