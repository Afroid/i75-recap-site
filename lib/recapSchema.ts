import { z } from "zod";

const RecapMatchupSchema = z.object({
  team1: z.string(),
  team2: z.string(),
  winner: z.string(),
  score: z.string(),
  breakdown: z.string(),
  imageUrl: z.string().optional(),
  imageAlt: z.string().optional(),
});

const GameNotesSectionSchema = z.object({
  type: z.literal("gameNotes"),
  pointsLeader: z.object({
    name: z.string(),
    points: z.number(),
    notes: z.string().optional(),
  }),
  dud: z.object({
    name: z.string(),
    points: z.number(),
    notes: z.string().optional(),
  }),
  matchups: z.array(RecapMatchupSchema),
});

const RecapSectionSchema = z.discriminatedUnion("type", [
  z.object({
    type: z.literal("intro"),
    content: z.string(),
  }),
  z.object({
    type: z.literal("tidbits"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("injuries"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("sidePieces"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("matchups"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("farewell"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  })
]);

const RecapWeekSchema = z.object({
  week: z.number(),
  title: z.string(),
  sections: z.array(z.union([RecapSectionSchema, GameNotesSectionSchema])),
});

export const RecapDataSchema = z.object({
  year: z.number(),
  recaps: z.array(RecapWeekSchema),
});
