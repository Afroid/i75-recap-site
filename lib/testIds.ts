export const TestIds = {
  /**
   * Pages
   */

  // Index - App Level
  SITE_HEADER: "site-header",
  MAIN_CONTENT: "main-content",
  PAGE_WRAPPER: "page-wrapper",

  // Home Page
  HOME_MAIN:   "home-main",
  HOME_HERO_LOGO: "home-hero-logo",
  HOME_INTRO_SECTION: "home-intro-section",
  HOME_HERO_HEADING: "home-hero-heading",
  HOME_CTA_CONTAINER: "home-cta-container",
  HOME_VIEW_ALL_RECAPS_BUTTON: "home-view-all-recaps-button",
  HOME_LEARN_MORE_BUTTON: "home-learn-more-button",

  // 404 Page
  NOT_FOUND_MAIN: "not-found-main",
  RETURN_HOME_LINK: "return-home-link",

  // About Page
  ABOUT_MAIN: "about-main",

  // Contact Page
  CONTACT_MAIN: "contact-main",
  CONTACT_EMAIL_LINK: "contact-email-link",
  CONTACT_FORM: "contact-form",
  CONTACT_FORM_NAME_SECTION: "contact-form-name-section",
  CONTACT_FORM_EMAIL_SECTION: "contact-form-email-section",
  CONTACT_FORM_MESSAGE_SECTION: "contact-form-message-section",
  CONTACT_FORM_SUBMIT: "contact-form-submit",

  /**
   * Recap Pages - (pages/recaps)
   */

  // Recap Page

  // View All Recaps Page
  VIEW_ALL_RECAPS: "view-all-recaps",
  // Year Page

  /**
   * Components
   */

  // Breadcrumb Component
  BREADCRUMB: "breadcrumb",
  BREADCRUMB_NAV: "breadcrumb-nav",
  BREADCRUMB_LIST: "breadcrumb-list",
  BREADCRUMB_ITEM: "breadcrumb-item",
  BREADRCRUMB_LINK: "breadcrumb-link",
  BREADCRUMB_CURRENT: "breadcrumb-current",
  BREADCRUMB_SEPARATOR: "breadcrumb-separator",

  // DesktopNavbar Component

  // GameNotesSection.tsx
  GAME_NOTES_SECTION: "game-notes-section",
  GAME_NOTES_ITEM: "game-notes-item",
  GAME_NOTES_SCOREBOARD: "game-notes-scoreboard",
  GAME_NOTES_BREAKDOWN: "game-notes-breakdown",
  GAME_NOTES_IMAGE_CONTAINER: "game-notes-image-container",
  GAME_NOTES_IMAGE: "game-notes-image",

  // Header.tsx
  HEADER_SITE_HEADER: "site-header",
  HEADER_MOBILE_HEADER: "mobile-header",
  HEADER_HAMBURGER_BUTTON: "header-hamburger-button",
  HEADER_MOBILE_LOGO: "header-mobile-logo",
  HEADER_DESKTOP_LOGO: "header-desktop-logo",
  HEADER_DESKTOP_NAV: "header-desktop-nav",
  HEADER_MOBILE_DRAWER: "header-mobile-drawer",

  // HeaderLogo.tsx
  HEADER_LOGO_LINK: "header-logo-link",
  HEADER_LOGO_BG_DARK: "header-logo-bg-dark",
  HEADER_LOGO_BG_LIGHT: "header-logo-bg-light",
  HEADER_LOGO_CONTAINER: "header-logo-container",
  HEADER_LOGO_NAME: "header-logo-name",
  HEADER_LOGO_DYNAMIC: "header-logo-dynamic",

  // LeaderDudSection.tsx
  LDS_SECTION: "lds-section",
  LDS_LEADER_HEADING: "lds-leader-heading",
  LDS_LEADER_POINTS: "lds-leader-points",
  LDS_LEADER_NOTES: "lds-leader-notes",
  LDS_DUD_HEADING: "lds-dud-heading",
  LDS_DUD_POINTS: "lds-dud-points",
  LDS_DUD_NOTES: "lds-dud-notes",

  // MobileDrawer.tsx
  MOBILE_SEASON_TOGGLE: "mobile-season-toggle",
  MOBILE_SEASON_WEEKS: "mobile-season-weeks",
  MOBILE_WEEK_LINK: "mobile-week-link",
  MOBILE_DRAWER_OVERLAY: "mobile-drawer-overlay",
  MOBILE_DRAWER: "mobile-drawer",
  MOBILE_DRAWER_TITLE: "mobile-drawer-title",
  MOBILE_OLDER_SEASON_TITLE: "mobile-older-season-title",
  MOBILE_TOGGLE_OLDER_SEASONS: "mobile-toggle-older-seasons",

  // RecapLayout.tsx
  RECAP_TITLE: "recap-title",
  RECAP_SECTION: "recap-section",
  RECAP_HEADING: "recap-heading",
  RECAP_INTRO: "recap-intro",
  RECAP_BULLETS: "recap-bullets",
  RECAP_BULLET: "recap-bullet",
  RECAP_CONTENT: "recap-content",
  RECAP_IMAGE: "recap-image",

  // Scoreboard.tsx
  SCOREBOARD_CONTAINER: "scoreboard-container",
  SCOREBOARD_ENTRY: "scoreboard-entry",
  SCOREBOARD_LOGO: "scoreboard-logo",
  SCOREBOARD_NAME: "scoreboard-name",
  SCOREBOARD_RECORD: "scoreboard-record",
  SCOREBOARD_SCORE: "scoreboard-score",

  // TeamLogo.tsx
  TEAM_LOGO_IMAGE: "team-logo-image",

  // Static Links
  NAV_LINK_HOME: "nav-link-home",
  NAV_LINK_ABOUT: "nav-link-about",
  NAV_LINK_CONTACT: "nav-link-contact"
}
