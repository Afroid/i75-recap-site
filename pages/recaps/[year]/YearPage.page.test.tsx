import fs from "fs";
import path from "path";
import React from "react";
import { render, screen } from "@testing-library/react";
import type { GetStaticPropsContext, GetStaticPathsContext } from "next";
import type { ParsedUrlQuery } from "querystring";
import { TestIds } from "@/lib/testIds";

import YearPage, {
  getStaticPaths,
  getStaticProps,
} from "./index.page";

// Stub framer-motion to render plain <li>
jest.mock("framer-motion", () => ({
  __esModule: true,
  motion: {
    li: ({
      children,
      ...props
    }: React.LiHTMLAttributes<HTMLLIElement>) => (
      <li {...props}>{children}</li>
    ),
  },
}));

// Mock fs and path so getStatic[Whatever] can be driven from tests
jest.mock("fs");
jest.mock("path");

// Helper to get proper types on our mocks
const mockedFs = fs as jest.Mocked<typeof fs>;
const mockedPath = path as jest.Mocked<typeof path>;

/**
 * Component tests
 * We render YearPage with a fake year + weeks prop and assert on the DOM plus snapshot it.
 */
describe("YearPage component", () => {
  it("renders a list of weeks with correct links", async () => {
    // Renders the page
    render(<YearPage year={2025} weeks={[1, 2]} />);

    // Waits for any hydration / layout effects using /Week 1/ as a regex
    await screen.findByRole("link", { name: /Week 1/ });

    const link1 = screen.getByTestId(`${TestIds.YEAR_WEEK_LINK}-1`);
    const link2 = screen.getByTestId(`${TestIds.YEAR_WEEK_LINK}-2`);
    expect(link1).toHaveAttribute("href", "/recaps/2025/week-1");
    expect(link2).toHaveAttribute("href", "/recaps/2025/week-2");

    // There should be 2 list items
    const items = screen.getAllByRole("listitem");
    expect(items).toHaveLength(2);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<YearPage year={2025} weeks={[1, 2, 3]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

/**
 * getStaticPaths test
 * Mocks a fake directory listing
 * Asserts on the returned paths array and fallback flag
 */
describe("getStaticPaths", () => {
  const FAKE_DIR = "/project/data/recaps";

  beforeEach(() => {
    // any call to path.join(...) returns our fake dir
    mockedPath.join.mockReturnValue(FAKE_DIR);

    // Pretends there is two .json files and one other file that isn't .json
    (fs.readdirSync as jest.Mock).mockReturnValue([
      "2023.json",
      "notes.txt",
      "2022.json",
    ]);
  });

  afterEach(() => jest.resetAllMocks());

  it("returns paths for each .json file and fallback=false", async () => {
    const result = await getStaticPaths({} as GetStaticPathsContext);

    // ensure we used the correct directory
    expect(path.join).toHaveBeenCalledWith(
      process.cwd(),
      "data",
      "recaps"
    );
    expect(result).toEqual({
      paths: [
        { params: { year: "2023" } },
        { params: { year: "2022" } },
      ],
      fallback: false,
    });
  });
});

/**
 * getStaticProps test
 * Mocks reading a JSON file
 * Asserts on the returned props object
 */
describe("getStaticProps", () => {
  const FAKE_FILE = "/project/data/recaps/2021.json";
  const SAMPLE_DATA = {
    year: 2021,
    recaps: [{ week: 10, title: "x", sections: [] }],
  };

  beforeEach(() => {
    // path.join(..., '2021.json') â†’ our fake file path
    mockedPath.join.mockReturnValue(FAKE_FILE);

    // Reading that file returns SAMPLE_DATA
    mockedFs.readFileSync.mockReturnValue(JSON.stringify(SAMPLE_DATA));
  });

  afterEach(() => jest.resetAllMocks());

  it("returns props with year and array of weeks", async () => {
    // Build a fake context object
    const context = {
      params: { year: "2021" },
    } as unknown as GetStaticPropsContext<ParsedUrlQuery>;

    const result = await getStaticProps(context);

    // Assertions
    expect(result).toEqual({
      props: {
        year: 2021,
        weeks: [10],
      },
    });

    expect(path.join).toHaveBeenCalledWith(
      process.cwd(),
      "data",
      "recaps",
      "2021.json"
    );
  });
});
