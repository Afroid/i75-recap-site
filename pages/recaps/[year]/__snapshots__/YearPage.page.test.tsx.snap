// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YearPage component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pt-4 max-w-2xl mx-auto"
    data-testid="year-page"
  >
    <div
      class="text-center"
    >
      <div
        class="inline-block"
      >
        <h1
          class="text-4xl lg:text-5xl font-extrabold text-gray-900 mb-2"
          data-testid="year-title"
        >
          2025 Season Recaps
        </h1>
        <div
          class="h-1 w-full bg-gradient-to-r from-green-300 to-green-700 rounded-full mb-6"
        />
      </div>
    </div>
    <ul
      class="grid grid-cols-1 sm:grid-cols-2 gap-6"
      data-testid="year-week-list"
    >
      <li
        animate="[object Object]"
        data-testid="year-week-item-1"
        initial="[object Object]"
        transition="[object Object]"
      >
        <a
          class="block rounded-lg shadow-md bg-white hover:bg-green-50 p-6 text-center transition border border-gray-200 hover:border-green-400 transform transition-transform hover:shadow-lg hover:scale-105 hover:-translate-y-1 flex flex-col justify-center min-h-[150px]"
          data-testid="year-week-link-1"
          href="/recaps/2025/week-1"
        >
          <h2
            class="text-xl font-semibold text-gray-900 mb-2"
          >
            Week 1
          </h2>
          <p
            class="text-gray-600"
          >
            View Recap
          </p>
        </a>
      </li>
      <li
        animate="[object Object]"
        data-testid="year-week-item-2"
        initial="[object Object]"
        transition="[object Object]"
      >
        <a
          class="block rounded-lg shadow-md bg-white hover:bg-green-50 p-6 text-center transition border border-gray-200 hover:border-green-400 transform transition-transform hover:shadow-lg hover:scale-105 hover:-translate-y-1 flex flex-col justify-center min-h-[150px]"
          data-testid="year-week-link-2"
          href="/recaps/2025/week-2"
        >
          <h2
            class="text-xl font-semibold text-gray-900 mb-2"
          >
            Week 2
          </h2>
          <p
            class="text-gray-600"
          >
            View Recap
          </p>
        </a>
      </li>
      <li
        animate="[object Object]"
        data-testid="year-week-item-3"
        initial="[object Object]"
        transition="[object Object]"
      >
        <a
          class="block rounded-lg shadow-md bg-white hover:bg-green-50 p-6 text-center transition border border-gray-200 hover:border-green-400 transform transition-transform hover:shadow-lg hover:scale-105 hover:-translate-y-1 flex flex-col justify-center min-h-[150px]"
          data-testid="year-week-link-3"
          href="/recaps/2025/week-3"
        >
          <h2
            class="text-xl font-semibold text-gray-900 mb-2"
          >
            Week 3
          </h2>
          <p
            class="text-gray-600"
          >
            View Recap
          </p>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
